.github/workflows/local_env_tests.yml

```diff
@@ -0,0 +1,60 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# To learn more about GitHub Actions in Apache Beam check the CI.md
+
+name: Local environment tests
+
+on:
+  push:
+    branches: ['master', 'release-*']
+    tags: 'v*'
+  pull_request:
+    branches: ['master', 'release-*']
+    tags: 'v*'
+    paths: ['sdks/python/**', 'model/**']
+  workflow_dispatch:
+    inputs:
+      runDataflow:
+        description: 'Type "true" if you want to run Dataflow tests (GCP variables must be configured, check CI.md)'
+        default: false
+
+jobs:
+  run_local_env_install_ubuntu:
+    timeout-minutes: 25
+    name: "Ubuntu run local environment shell script"
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: "Installing local env dependencies"
+        run: "sudo ./local-env-setup.sh"
+        id: local_env_install_ubuntu
+      - name: "Gradle check"
+        run: "./gradlew checkSetup"
+        id: local_env_install_gradle_check_ubuntu
+  run_local_env_install_mac:
+    timeout-minutes: 50
+    name: "Mac run local environment shell script"
+    runs-on: macos-latest
+    steps:
+      - uses: actions/checkout@v2
+      - name: "Installing local env dependencies"
+        run: "./local-env-setup.sh"
+        id: local_env_install_mac
+      - name: "Gradle check"
+        run: "./gradlew checkSetup"
+        id: local_env_install_gradle_check_mac
```

Comment:
Have these been tested?

Reply:
Hi Brian! Yeah, this has been tested successfully.